### PR TITLE
#2 refactor architecture docs

### PR DESCRIPTION
**Pull request by lucksus**
I'm creating a PR for this ticket already although we are still missing two ADRs.

I've checked off the boxes in the ticket description for those ADRs that I feel a mature enough to get review on. The 64bit one is still empty. The last one for DBs is still missing.

Anyway, please have a look guys :)

### CODE REVIEW

**Comment by zippy** [comment]
I updated the 64bit one.

File: doc/architecture/decisions/0002-rewrite-second-iteration-in-rust.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+There is a WASM interpreter in Rust (WASMI).
+
+Only problem might be problems integration with Promether / linking against C++:
+* https://doc.rust-lang.org/nomicon/ffi.html
+* https://users.rust-lang.org/t/linking-with-custom-c-library/637
+
+
+## Decision
+
+Refactor Holochain in the Rust programming language
```
Comment (ddd-mtl) [inline doc/architecture/decisions/0002-rewrite-second-iteration-in-rust.md]: Change 'Refactor' to 'Rewrite'

File: doc/architecture/decisions/0003-redux-architecture-pattern.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Date: 2018-05-16
+
+## Status
+
+Accepted
+
+## Context
+
+We are doing a rebuild.
```
Comment (ddd-mtl) [inline doc/architecture/decisions/0003-redux-architecture-pattern.md]: Change to: We are doing a rewrite (see see ADR:0002)

File: doc/architecture/decisions/0003-redux-architecture-pattern.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+## Decision
+
+The new holochain architecture will follow a redux architecture in order for an agent to have one global state.
+
+We will apply nested state objects which represent a state tree, with sub states for each module.
+
+We use reference counting smart pointers for the sub
+states such that it is possible for each module's
+reducer to decide if the sub state is to be mutated or reused.
```
Comment (ddd-mtl) [inline doc/architecture/decisions/0003-redux-architecture-pattern.md]: Remove newlines

File: doc/architecture/decisions/0005-alpha1-is-last-major-go-release.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+Date: 2018-05-16
+
+## Status
+
+Accepted
+
+## Context
+
+A complete Rust rewrite is planned (see ADR #0)
```
Comment (ddd-mtl) [inline doc/architecture/decisions/0005-alpha1-is-last-major-go-release.md]: ADR:0002

File: doc/architecture/decisions/0006-splitting-agent-into-front-house-back-house-or-not.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# 6. Splitting agent into front-house / back-house - OR NOT
```
Comment (ddd-mtl) [inline doc/architecture/decisions/0006-splitting-agent-into-front-house-back-house-or-not.md]: Will we change the filename once we change the status to Accepted or Rejected?

File: doc/architecture/decisions/0011-build-hcadmin-app-with-qt-c.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+## Status
+
+proposed
+
+## Context
+
+In the alpha version we had *hcd* that runs a single Holochain app/agent as a background demon.
+
+Going forward we need a more sophisticated container of several Holochain apps for several reasons:
+* UX
```
Comment (ddd-mtl) [inline doc/architecture/decisions/0011-build-hcadmin-app-with-qt-c.md]: One end-user friendly tool to administer all installed and running Holochain apps.

File: doc/architecture/decisions/0011-build-hcadmin-app-with-qt-c.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+* Mid-term roadmap entails having a generalized Holochain UI / browser that would act as an app container as well
+
+Since with ADR 8 we want to go mobile first, we need to have an easy way to run Holochain apps on mobile phones.
+
+Qt sports QML for rapid UI development and compiles the same code natively to Windows/Linux/MacOS/Android/iOS/Blackberry.
+For system and network capabilities, C++ would be at our disposal.
+
+## Decision
+
+Use Qt framework for building a cross platform Holochain app container as a replacement for *hcadmin* for Holochain app managament & deployment, and possiblly integrating *hcd* into it if we can manage the Rust integration.
```
Comment (ddd-mtl) [inline doc/architecture/decisions/0011-build-hcadmin-app-with-qt-c.md]: "managament " Typo

File: doc/architecture/decisions/0011-build-hcadmin-app-with-qt-c.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+## Consequences
+
+* We could have a system tray app to manage/start/stop HC apps.
+* We could have the exact same UI on all relevant platforms including mobile with only one code base.
+* -> a mobile app would be easy to build.
+* C++ code needs to be maintained.
```
Comment (ddd-mtl) [inline doc/architecture/decisions/0011-build-hcadmin-app-with-qt-c.md]: Cross-platform C++ code

File: doc/architecture/decisions/0011-build-hcadmin-app-with-qt-c.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+* UI/UX needs to be considered re conforming to different platform standards vs. creating one HC design & feel.
```
Comment (ddd-mtl) [inline doc/architecture/decisions/0011-build-hcadmin-app-with-qt-c.md]: "re" ?

File: doc/architecture/decisions/0003-redux-architecture-pattern.md (hunk 2 above)
Comment (lucksus) [reply doc/architecture/decisions/0003-redux-architecture-pattern.md]: Am changing all of these right, though: I perceive rewrite and rebuild (in this context) to have the same meaning.

File: doc/architecture/decisions/0006-splitting-agent-into-front-house-back-house-or-not.md (hunk 5 above)
Comment (lucksus) [reply doc/architecture/decisions/0006-splitting-agent-into-front-house-back-house-or-not.md]: I would not. Why should we?

**Comment by lucksus** [comment]
@ddd-mtl, can you check again? I've implemented your change requests.

**Pull request merged**